### PR TITLE
Fix Migration not support PostgreSQL json type

### LINKED ISSUE #24023: Migration not support PostgreSQL json type.
## Bug Report

Migration do not support PostgreSQL json type.

source table init SQL
```
CREATE TABLE test.t_order (order_id int8 NOT NULL,user_id int4 NOT NULL,status varchar ( 50 ) NULL,t_json json NULL);

INSERT INTO test.t_order (order_id, user_id, status, t_json) VALUES (1, 2, '1', null);
```

When execute `MIGRATE TABLE ds_0.test.t_order INTO sharding_db.t_order;`, wait the job status is `EXECUTE_INVENTORY_TASK`.
Then execute sql `UPDATE test.t_order SET t_json = '{"test": "ok2"}' WHERE order_id = 1;` at source datatase

The following error will occur
```
ERROR: column "t_json" is of type json but expression is of type character varying
  建议：You will need to rewrite or cast the expression.
  位置：79  Call getNextException to see other errors in the batch.
```

### Which version of ShardingSphere did you use?

master

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

ShardingSphere-Proxy

### Expected behavior

No exception occur and data is correctly synchronized

### Actual behavior

exception occur

### PR DESCRIPTION
**Pull request by azexcy**
Fixes #24023.

Changes proposed in this pull request:
  - Fix Migration not support PostgreSQL json type
  - Pipeline E2E update `t_json` column value

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by azexcy** [comment]
reference documents
```
https://jdbc.postgresql.org/documentation/use/
```

File: test/e2e/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/task/PostgreSQLIncrementTask.java [hunk 1]
```diff
@@ -73,8 +73,8 @@ public void run() {
     private Object insertOrder() {
         ThreadLocalRandom random = ThreadLocalRandom.current();
         String status = 0 == random.nextInt() % 2 ? null : "中文测试";
-        Object[] orderInsertDate = new Object[]{KEY_GENERATE_ALGORITHM.generateKey(), random.nextInt(0, 6), status};
-        String insertSQL = String.format("INSERT INTO %s (order_id,user_id,status) VALUES (?, ?, ?)", getTableNameWithSchema(orderTableName));
+        Object[] orderInsertDate = new Object[]{KEY_GENERATE_ALGORITHM.generateKey(), random.nextInt(0, 6), status, PipelineCaseHelper.generateJsonString(2)};
+        String insertSQL = String.format("INSERT INTO %s (order_id,user_id,status,t_json) VALUES (?, ?, ?, ?)", getTableNameWithSchema(orderTableName));
```
Comment (sandynz) [inline test/e2e/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/task/PostgreSQLIncrementTask.java:77]: It's better to support `jsonb` at this time

**Review by sandynz — CHANGES_REQUESTED** [review]

File: test/e2e/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/task/PostgreSQLIncrementTask.java [hunk 2]
```diff
@@ -73,8 +73,9 @@ public void run() {
     private Object insertOrder() {
         ThreadLocalRandom random = ThreadLocalRandom.current();
         String status = 0 == random.nextInt() % 2 ? null : "中文测试";
-        Object[] orderInsertDate = new Object[]{KEY_GENERATE_ALGORITHM.generateKey(), random.nextInt(0, 6), status, PipelineCaseHelper.generateJsonString(2)};
-        String insertSQL = String.format("INSERT INTO %s (order_id,user_id,status,t_json) VALUES (?, ?, ?, ?)", getTableNameWithSchema(orderTableName));
+        String jsonString = PipelineCaseHelper.generateJsonString(2);
+        Object[] orderInsertDate = new Object[]{KEY_GENERATE_ALGORITHM.generateKey(), random.nextInt(0, 6), status, jsonString, jsonString.getBytes()};
```
Comment (sandynz) [inline test/e2e/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/task/PostgreSQLIncrementTask.java:77]: 1, `jsonString.getBytes()`, could we support Unicode character?

2, It's better to make `PipelineCaseHelper.generateJsonString(2)` could generate Unicode characters.

**Review by sandynz — CHANGES_REQUESTED** [review]

File: test/e2e/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/task/PostgreSQLIncrementTask.java (hunk 2 above)
Comment (azexcy) [reply test/e2e/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/task/PostgreSQLIncrementTask.java:77]: I find data consistency check failed, i need takes a little time to check

**Comment by azexcy** [comment]
Data consistency checks will fail when only unique keys are found, fixed at this pr

File: test/e2e/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/base/PipelineBaseE2EIT.java [hunk 3]
```diff
@@ -201,9 +201,13 @@ protected void addResource(final String distSQL) throws SQLException {
...
-        return DatabaseTypeUtil.isMySQL(getDatabaseType())
-                ? new JdbcUrlAppender().appendQueryProperties(jdbcUrl, PropertiesBuilder.build(new Property("rewriteBatchedStatements", Boolean.TRUE.toString())))
-                : jdbcUrl;
+        if (DatabaseTypeUtil.isMySQL(getDatabaseType())) {
+            return new JdbcUrlAppender().appendQueryProperties(jdbcUrl, PropertiesBuilder.build(new Property("rewriteBatchedStatements", Boolean.TRUE.toString())));
+        }
+        if (DatabaseTypeUtil.isPostgreSQL(getDatabaseType()) || DatabaseTypeUtil.isOpenGauss(getDatabaseType())) {
+            return new JdbcUrlAppender().appendQueryProperties(jdbcUrl, PropertiesBuilder.build(new Property("stringtype", "unspecified")));
+        }
+        return jdbcUrl;
```
Comment (sandynz) [inline test/e2e/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/base/PipelineBaseE2EIT.java:210]: `jdbcUrl` should be `result`

File: test/e2e/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/framework/helper/PipelineCaseHelper.java [hunk 4]
```diff
@@ -102,8 +102,21 @@ private static String generateString(final int strLength) {
...
-    private static String generateJsonString(final int strLength) {
-        return String.format("{\"test\":\"%s\"}", generateString(strLength));
+    /**
+     * Generate json string.
+     *
+     * @param useUnicodeCharacter use unicode character
+     * @param strLength string length
+     * @return json string
+     */
+    public static String generateJsonString(final int strLength, final boolean useUnicodeCharacter) {
```
Comment (sandynz) [inline test/e2e/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/framework/helper/PipelineCaseHelper.java:112]: `strLength` could be `length`

**Review by sandynz — CHANGES_REQUESTED** [review]

**Comment by sandynz** [comment]
Unicode character support could be done in next PR

**Review by sandynz — APPROVED** [review]

**Pull request merged**